src/component/editor/compose/index.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { useMemo, useState } from "react";
+import { createEditor } from "slate";
+import { withHistory } from "slate-history";
+import { Slate, withReact } from "slate-react";
+import {
+  ParagraphPlugin,
+  HeadingPlugin,
+  BoldPlugin,
+  EditablePlugins,
+  ItalicPlugin,
+  UnderlinePlugin,
+  CodePlugin,
+  StrikethroughPlugin,
+  BlockquotePlugin,
+  ListPlugin,
+  CodeBlockPlugin,
+  ResetBlockTypePlugin,
+  SoftBreakPlugin,
+  ExitBreakPlugin,
+  MentionPlugin,
+  pipe,
+  SlateDocument,
+  withAutoformat,
+  withImageUpload,
+  withLink,
+  withList,
+  withMarks,
+  withInlineVoid,
+  MentionSelect,
+  useMention,
+} from "@udecode/slate-plugins";
+import {
+  headingTypes,
+  options,
+  optionsResetBlockTypes,
+  initialValueEmpty,
+} from "component/editor/config/initialValues";
+import { autoformatRules } from "component/editor/config/autoformatRules";
+import { MENTIONABLES } from "component/editor/config/mentionables";
+
+const plugins = [
+  ParagraphPlugin(options),
+  BoldPlugin(),
+  UnderlinePlugin(),
+  ItalicPlugin(),
+  CodePlugin(),
+  StrikethroughPlugin(),
+  BlockquotePlugin(options),
+  ListPlugin(options),
+  HeadingPlugin(options),
+  CodeBlockPlugin(options),
+  ResetBlockTypePlugin(optionsResetBlockTypes),
+  SoftBreakPlugin({
+    rules: [
+      { hotkey: "shift+enter" },
+      {
+        hotkey: "enter",
+        query: {
+          allow: [options.code_block.type, options.blockquote.type],
+        },
+      },
+    ],
+  }),
+  ExitBreakPlugin({
+    rules: [
+      {
+        hotkey: "mod+enter",
+      },
+      {
+        hotkey: "mod+shift+enter",
+        before: true,
+      },
+      {
+        hotkey: "enter",
+        query: {
+          start: true,
+          end: true,
+          allow: headingTypes,
+        },
+      },
+    ],
+  }),
+  MentionPlugin(options),
+];
+
+interface EditorProps {}
+
+const withPlugins = [
+  withReact,
+  withHistory,
+  withLink(),
+  withList(options),
+  withMarks(),
+  withImageUpload(),
+  withAutoformat({ rules: autoformatRules }),
+  withInlineVoid({ plugins }),
+] as const;
+
+const ComposeEditor = (props: EditorProps) => {
+  const [value, setValue] = useState(initialValueEmpty);
+  const editor = useMemo(() => pipe(createEditor(), ...withPlugins), []);
+  const {
+    onAddMention,
+    onChangeMention,
+    onKeyDownMention,
+    search,
+    index,
+    target,
+    values,
+  } = useMention(MENTIONABLES, {
+    maxSuggestions: 10,
+    trigger: "@",
+  });
+
+  return (
+    <Slate
+      editor={editor}
+      value={value}
+      onChange={(newValue: any) => {
+        setValue(newValue as SlateDocument);
+
+        //trigger mention
+        onChangeMention(editor);
+
+        //save to local storage for to persist...
+        const content = JSON.stringify(value);
+        localStorage.setItem("content", content);
```

Comment:
I would like to better understand what this local storage operation is about.

Reply:
As someone types into the editor, we save the state in localStorage aka web storage. This should mean that if someone refreshes the page in the middle of typing a message we save the text they've already written in localStorage.

This isn't enable yet. I'll make a new pull request to enable.